src/textual_fspicker/select_directory.py

```diff
@@ -91,23 +61,61 @@ def __init__(
 
     def on_mount(self) -> None:
         """Configure the dialog once the DOM is ready."""
+        super().on_mount() # Call parent's on_mount
         navigation = self.query_one(DirectoryNavigation)
         navigation.show_files = False
-        self.query_one(CurrentDirectory).current_directory = navigation.location
+
+        path_input = self.query_one("#path_input", Input)
+        path_input.value = str(navigation.location)
+        # navigation.focus() # Focus is handled by super().on_mount or should be reconsidered
```

Comment:
This sort of comment shouldn't be necessary. If something like this isn't obvious and would be confusing if it wasn't here it might indicate a problem with the approach being used.